feat(generic): save cookie with form data

b1rger opened the pull request:
Closes: #1564

b1rger committed:
feat(generic): save cookie with form data

With this feature, the form data of the List view is saved as a cookie
and reused once the List view is visited again. The 'reset' button still
resets the form and now also deletes the cookie.

gythaogg commented on apis_core/generic/views.py:
```diff
@@ -126,6 +126,23 @@ class List(
+    def setup(self, *args, **kwargs):
+        super().setup(*args, **kwargs)
+        content_type = ContentType.objects.get_for_model(self.model)
+        self.cookiename = f"{content_type.app_label}.{content_type.name}_list"
```
This can result in illegal cookie names if  `content_type.name` contains spaces or other delimiters that are [forbidden](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Set-Cookie#:~:text=A%20%3Ccookie%2Dname%3E%20can%20contain%20any%20US%2DASCII%20characters%20except%20for%20control%20characters%20(ASCII%20characters%200%20up%20to%2031%20and%20ASCII%20character%20127)%20or%20separator%20characters%20(space%2C%20tab%20and%20the%20characters%3A%20(%20)%20%3C%20%3E%20%40%20%2C%20%3B%20%3A%20%5C%20%22%20/%20%5B%20%5D%20%3F%20%3D%20%7B%20%7D))
For example: 
<img width="1110" height="261" alt="image" src="https://github.com/user-attachments/assets/e3a49247-4bb2-4707-a417-4edd13061029" />

gythaogg reviewed (changes requested):
Works well when the list view names do not contain spaces or special characters. Thanks!

Use `content_type.model` instead of `content_type.name` to avoid illegal cookie names

gythaogg reviewed (approved):
Works!
This might drastically change the user experience though - if projects do not want this feature, it would be nice to have the possibility to disable it via a django setting. 

Other options might be to have the option to remember/not the form data as 
1. a user profile setting in the database
2. a session setting stored in cookies

b1rger committed:
feat(generic): add a reset checkbox to column and filterset forms

The checkbox in the columns form is part of the form class. The one
for the filterset has to be injected outside of the form class,
otherwise the filterset gets confused. So we simply put it directly in
the template.
